File: admin/src/api/interface/sysRole.ts
// sys_role 角色
export interface ReqSysRoleParams extends RequestPage {
  code?: string // 编码
  deleted?: number // 删除:0否/1是
  status?: number // 状态:0正常/1停用
  name?: string // 名称
}

export interface ResSysRole {
  id: number // 编号
  name: string // 名称
  code: string // 编码
  scope: number | undefined // 数据范围:1:全部数据权限/2:自定数据权限/3:本部门数据权限/4:本部门及以下数据权限
  scopeDept: any | undefined // 数据范围(指定部门数组)
  sort: number // 排序
  status: number // 状态:0正常/1停用
  deleted: number // 删除:0否/1是
  tenantId: number // 租户
  creator: string | undefined // 创建人
  createTime: string | undefined // 创建时间
  updater: string | undefined // 更新人
  updateTime: string | undefined // 更新时间
  menuIds: any | undefined // 菜单编号
}

File: cloud/dao/sys_role.go
package dao

import "github.com/abulo/ratel/v3/stores/null"

// SysRole 角色 sys_role
type SysRole struct {
	Id         *int64        `gorm:"primaryKey;autoIncrement;column:id" json:"id"` //bigint 编号,PRI
	Name       *string       `gorm:"column:name" json:"name"`                      //varchar 名称
	Code       *string       `gorm:"column:code" json:"code"`                      //varchar 编码
	Scope      null.Int32    `gorm:"column:scope" json:"scope"`                    //tinyint 数据范围:1:全部数据权限/2:自定数据权限/3:本部门数据权限/4:本部门及以下数据权限
	ScopeDept  null.JSON     `gorm:"column:scope_dept" json:"scopeDept"`           //json 数据范围(指定部门数组)
	Sort       *int32        `gorm:"column:sort" json:"sort"`                      //int 排序
	Status     *int32        `gorm:"column:status" json:"status"`                  //tinyint 状态:0正常/1停用
	Deleted    *int32        `gorm:"column:deleted" json:"deleted"`                //tinyint 删除:0否/1是
	TenantId   *int64        `gorm:"column:tenant_id" json:"tenantId"`             //bigint 租户
	Creator    null.String   `gorm:"column:creator" json:"creator"`                //varchar 创建人
	CreateTime null.DateTime `gorm:"column:create_time" json:"createTime"`         //datetime 创建时间
	Updater    null.String   `gorm:"column:updater" json:"updater"`                //varchar 更新人
	UpdateTime null.DateTime `gorm:"column:update_time" json:"updateTime"`         //datetime 更新时间
	MenuIds    null.JSON     `gorm:"column:menu_ids;<-:false" json:"menuIds"`      //菜单
}

func (SysRole) TableName() string {
	return "sys_role"
}
